torch/jit/__init__.py

```diff
@@ -900,10 +900,65 @@ def _try_compile_weak_script(fn):
         _jit_internal.compiled_weak_fns[fn]["compiled_fn"] = compiled_fn
         entry["status"] = _jit_internal.COMPILED
         return compiled_fn
+        # TODO: use fn.__closure__
+        raise RuntimeError("Cannot make resolutionCallback in Python 2")
     else:
         return entry["compiled_fn"]
 
 
+class ScriptWarning(Warning):
+    pass
+
+
+def make_rcb(fn):
+    """
+    Create a resolutionCallback by introspecting the function instead of
+    looking up the stack for the enclosing scope
+    """
+    if PY2:
+        # TODO: use fn.__closure__
+        raise RuntimeError("Cannot make resolutionCallback in Python 2")
+    else:
+        closure = inspect.getclosurevars(fn)
+    f_globals = closure.globals
+    f_unbound = closure.unbound
+    f_builtins = closure.builtins
+
+    def env(key):
```

Comment:
It should be possible to do this correctly, including free variables. fn.__code__.co_freevars (or something like that) holds the free variable names, and then `fn.<something else I forget with the name slot in it>` has the slots that correspond to the free variables. We should have tests that ensure resolution order here is the same  as in the real case (i.e. with shadowing globals). Free variables should come before globals. 

I am not sure we need inspect.getclosurevars I think the `__code__` object, etc. have enough info in them.